tips/TIP-0024/tip-0024.md

```diff
@@ -0,0 +1,178 @@
+- `Protocol Version` must match the `Protocol Version` config parameter of the node.
+- Parents:
+  - `Parents Count` must be at least 1 and not larger than `Max Parents Count`.
+  - `Parents` must be sorted in lexicographical order.
+  - Each `Block ID` must be unique.
+- Payload (if present):
+  - `Payload Type` must match one of the values described under [Payloads](#payloads).
+  - `Data Fields` must be correctly parsable in the context of the `Payload Type`.
+  - The payload itself must pass syntactic validation.
+- `Nonce` must be valid with respect to the PoW condition described under [Payloads](#payloads). The PoW score itself is computed according to [TIP-12](../TIP-0012/tip-0012.md).
+- There must be no trailing bytes after all block fields have been parsed.
+  
+### PoW validation
+
+The PoW that needs to be performed for each block protects the network against denial-of-service attacks where in a short time too many blocks are issued for the nodes to process. As the processing time of a block heavily depends on the contained payload, the PoW check can also depend on the `Payload Type` and is described under [Payloads](#payloads).
+It is important to note, that the actual parsing and validating of a payload can be computationally expensive. Thus, it is recommended to first parse the block with all its fields including `Payload Type` (but not parsing or validating the actual payload `Data Fields`). Now, simple syntactic validation steps &ndash; including PoW validation &ndash; can be performed and invalid blocks already filtered out before the payload is validated. With this approach, payload-based PoW validation is not significantly more expensive than payload-agnostic validation.
+
+## Payloads
+
+While blocks without a payload, i.e. `Payload Length` set to zero, are valid, such blocks do not contain any information. As such, blocks usually contain a payload. The detailed specification of each payload type is out of scope of this TIP. The following table lists all currently specified payloads that can be part of a block and links to their specification:
+
+| Payload Name | Type Value | TIP                               | PoW Condition                  |
+| ------------ | ---------- | --------------------------------- | ------------------------------ |
+| No Payload   | -          | -                                 | PoW score ≥ `Min PoW Score`    |
+| Tagged Data  | 5          | [TIP-23](../TIP-0023/tip-0023.md) | PoW score ≥ `Min PoW Score`    |
+| Transaction  | 6          | [TIP-20](../TIP-0020/tip-0020.md) | PoW score ≥ `Min PoW Score`    |
+| Milestone    | 7          | [TIP-29](../TIP-0029/tip-0029.md) | `nonce` = `0x0000000000000000` |
+
+## Example
+
+Below is the full serialization of a valid block with a _Tagged Data Payload_. The tag is the "IOTA" ASCII string and the data is the "hello world" ASCII string. Bytes are expressed as hexadecimal numbers.
+
+- Protocol Version (1-byte): `00` (0)
```

Comment:
Should probably put protocol version 2, I don't think this block would have been valid with version 0